### DISCUSSION TITLE
The GEdgePlacement....

### DISCUSSION THREAD

**Discussion by rsoika** [eclipse-glsp/glsp#863] [Q&A] [answered]
Hi,
can someone explain the `GEdgePlacement` a little bit. This Seems to be a powerful concept but I could not find any documentation and also the source code is not commented at all :-(

**Comment by planger** [comment, marked as answer]
Hi,

there is a bit of documentation available, but admittedly not much:
https://www.eclipse.org/glsp/documentation/clientlayouting/#edge-layout

The edge placement specifies how a child of an edge is placed relative to the edge. You can specify a side (left, right, top, bottom, on), an offset, and a position, which I think is in percent (0.5 means center, 0 means at the source, 1 means at the target). Additionally you can specify whether to rotate the child according to the angle of the edge.

See also API for the builder:
https://github.com/eclipse-glsp/glsp-server/blob/master/plugins/org.eclipse.glsp.graph/src/org/eclipse/glsp/graph/builder/impl/GEdgePlacementBuilder.java

I hope this helps!

**Reply by rsoika** [reply]
Thanks!
